lib/rbUtil.js

```diff
@@ -207,6 +207,22 @@ rbUtil.parseETag = function(etag) {
     }
 };
 
+/**
+ * Copies forwarded headers from restbase to request.
+ * If the same header was already set it takes precedence over
+ * the forwarded header.
+ */
+rbUtil.copyForwardedCookies = function(restbase, req) {
+    if (restbase._rootReq
```

Comment:
Do you think it would be worth extracting the cookie once, when assigning a new rootreq?